### ISSUE TITLE
Training accuracy stuck at 6% and loss at 2.7 using LSTM

### ISSUE THREAD

**Issue by srijandas07** [keras-team/keras#8190]
I am using LSTM for action recognition. My basic LSTM implemented on  Keras is getting accuracy of 76% on test data on one dataset(CAD60) but  when I use other dataset, my model gets stuck at a loss. Its predicting a  single class always. What can be the problem since I am using the exact framework, features  on both the dataset. Even I tried to tune the learning rate change the  optimizer but it didn't worked.

model = Sequential() # input has shape (samples, timesteps, locations)
model.add(LSTM(128, batch_input_shape=(batch_size, timesteps, data_dim)))
model.add(Dense(num_classes, activation='softmax'))

model.compile(loss='categorical_crossentropy',
          optimizer='adam',
          metrics=['accuracy'])

**Comment by PForet** [comment]
Are your classes unbalanced ? If so, accuracy is not a good metric, as predicting the most frequent class is a good way to have a high accuracy without actually learning anything. A solution could be to set weights to your observation, such as each class has the same total weight in your dataset.
See [here](https://datascience.stackexchange.com/questions/13490/how-to-set-class-weights-for-imbalanced-classes-in-keras ) for implementation.

**Comment by srijandas07** [comment]
Since I am using very well known public datasets, the problem of class imbalance is not there.
But yes the number of available sequences for classification is though less.

**Comment by PForet** [comment]
There are lots of very well knows public datasets with class imbalance. Can we know on which one you experience this issue ?

**Comment by srijandas07** [comment]
I experience this issue on MSRDailyActivity3D, which is quite well balanced.
Though I solved this issue using TimeDistributed layer at the end of LSTM.
But now the problem I am facing is overfitting. My accuracy on test data is very poor. I tried regularizer, adding Gaussian Noise, Gaussian Drop out but the problem persists.

**Comment by AndrasEros** [comment]
Just a tip, you may find interesting to try Phased-LSTM that tends not to overfit so much.

https://github.com/fferroni/PhasedLSTM-Keras
